tools/serverpod_cli/lib/src/analyzer/entities/validation/restrictions.dart

```diff
@@ -169,7 +173,46 @@ class Restrictions {
     return [];
   }
 
+  List<SourceSpanSeverityException> validateRelationInterdependencies(
+    String parentNodeName,
+    dynamic content,
+    SourceSpan? span,
+  ) {
+    var errors = <SourceSpanSeverityException>[];
+    var definition = documentDefinition;
+
+    if (definition is! ClassDefinition) return [];
```

Comment:
NIT (non-blocking): Return errors here since we have already created it, alternatively create errors below this line.

Reply:
There shouldn't be any errors if the definition is not a ClassDefinition (relations cannot be set on exceptions or enums). I have this here as an early return.

Reply:
Sorry, bad phrasing. Return the variable `errors` that is defined on line 181 in the early return or define it below this line.

Reply:
Ahh! Okay got it!